src/vcpkg/base/system.cpp

```diff
@@ -50,11 +70,34 @@ namespace vcpkg
     CPUArchitecture get_host_processor()
     {
 #if defined(_WIN32)
-        auto w6432 = get_environment_variable("PROCESSOR_ARCHITEW6432");
-        if (const auto p = w6432.get()) return to_cpu_architecture(*p).value_or_exit(VCPKG_LINE_INFO);
+        auto raw_w6432 = get_environment_variable("PROCESSOR_ARCHITEW6432");
+        if (const auto w6432 = raw_w6432.get())
+        {
+            const auto parsed_w6432 = to_cpu_architecture(*w6432);
+            if (const auto parsed = parsed_w6432.get())
+            {
+                return *parsed;
+            }
+
+            msg::print(Color::warning, msgProcessorArchitectureW6432Malformed, msg::value = *w6432);
+        }
+
+        const auto raw_processor_architecture = get_environment_variable("PROCESSOR_ARCHITECTURE");
+        const auto processor_architecture = raw_processor_architecture.get();
+        if (!processor_architecture)
+        {
+            msg::print(Color::error, msgProcessorArchitectureMissing);
+            Checks::exit_with_code(VCPKG_LINE_INFO, 1);
```

Comment:
I asked @strega-nil if this is the intended usage for now.

Reply:
```suggestion
            Checks::exit_with_message(VCPKG_LINE_INFO, msgProcessorArchitectureMissing);
```